src/components/Map.vue

```diff
@@ -276,15 +275,39 @@ export default {
         });
       },
     },
+    isMrXVisible: {
+      handler() {
+        const svg = document.getElementById("board");
+        if (this.isMrXVisible) {
+          // check if user has access to Mr X
+          const isMrXTurn =
+            this.currentPlayer && this.currentPlayer.role === "mr-x";
+
+          if (!isMrXTurn && this.isMrXVisible && this.mrXLocation) {
+            const mrXLocationStation = this.$refs[this.mrXLocation][0];
+            const attributes = {
+              cx: mrXLocationStation.getAttributeNS(null, "cx"),
+              cy: mrXLocationStation.getAttributeNS(null, "cy"),
+              className: `mr-x-location`,
+            };
+            this.addCircleToBoard(attributes);
+          }
+        } else {
```

Comment:
So this portion assumes that is a previous marker on the map for Mr X and you are now removing it. So for example on move # 4, you would enter the else portion of the if - else conditional, and remove the X marker that was added in move # 3

Reply:
exactly, just in case there was a marker, we want to remove it - this is particularly helpful if Mr. X decides to make a 2x move after his reveal move.